internal/nodes/metadata.go

```diff
@@ -0,0 +1,119 @@
+package nodes
+
+import (
+	"strings"
+
+	"github.com/berops/claudie/proto/pb"
+
+	k8sV1 "k8s.io/api/core/v1"
+)
+
+type LabelKey string
+
+const (
+	Nodepool         LabelKey = "claudie.io/nodepool"
+	Provider         LabelKey = "claudie.io/provider"
+	ProviderInstance LabelKey = "claudie.io/provider-instance"
+	NodeType         LabelKey = "claudie.io/node-type"
+	KubernetesZone   LabelKey = "topology.kubernetes.io/zone"
+	KubernetesRegion LabelKey = "topology.kubernetes.io/region"
+	KubernetesOs     LabelKey = "kubernetes.io/os"
+	KubernetesArch   LabelKey = "kubernetes.io/arch"
+	KubeoneOs        LabelKey = "v1.kubeone.io/operating-system"
+)
+
+const (
+	ControlPlane = "node-role.kubernetes.io/control-plane"
+	NoSchedule   = "NoSchedule"
+)
+
+// GetAllLabels returns default labels with their theoretical values for the specified nodepool.
+func GetAllLabels(np *pb.NodePool) map[string]string {
+	m := make(map[string]string)
+	// Claudie assigned labels.
+	m[string(Nodepool)] = np.Name
+	m[string(Provider)] = np.GetDynamicNodePool().Provider.CloudProviderName
+	m[string(ProviderInstance)] = np.GetDynamicNodePool().Provider.SpecName
+	m[string(NodeType)] = getNodeType(np)
+	m[string(KubernetesZone)] = sanitiseString(np.GetDynamicNodePool().Zone)
+	m[string(KubernetesRegion)] = sanitiseString(np.GetDynamicNodePool().Region)
+	// Other labels.
+	m[string(KubernetesOs)] = "linux"   // Only Linux is supported.
+	m[string(KubernetesArch)] = "amd64" // TODO add arch https://github.com/berops/claudie/issues/665
+	m[string(KubeoneOs)] = "ubuntu"     // Only supported Os
+
+	// Add custom user defined labels.
+	for k, v := range np.Labels {
+		m[k] = v
+	}
```

Comment:
What if someones overwrites some of our default labels?

Reply:
Interesting, what would you suggest we do in this case?

Reply:
First build custom labels, and later overwrite the map with "well-know" labels, to ensure that they won't get changed. WDYT?

Reply:
I would actually go the other way around. The claudie default labels are quite specific, so if the customer wishes to overwrite them, I think they will have a reason for that. Can't imagine a case where they would mistakenly overwrite them.

Reply:
In the internal communication it was agreed we will keep default labels even if the user will try to overwrite them.